### DISCUSSION TITLE
Communication flow from ground station to the components cFS app

### DISCUSSION THREAD

**Discussion by StefanoSWE** [nasa/nos3#232] [Q&A]
Hello everybody.
I'm a new user of NOS3 and I'd like to have support on the following situation.
From [NOS3 Architecture Wiki page](https://github.com/nasa/nos3/wiki/NOS3-Architecture) , I see that Ground SW communicates with FSW and this, through NOS Engine busses, can exchange messages with HW simulators.
Since from this other page [NOS3-Components,-Repository-and-Directory-Structure](https://github.com/nasa/nos3/wiki/NOS3-Components,-Repository-and-Directory-Structure) I read that a NOS3 component (components folder) is defined by FSW (considered as cFS application), GSW and SIM (which simulates the HW behavior), does the Flight Software system in the Architecture view include also the component app (components/xxx/fsw) and so this app can communicate with simulated HW through a specific bus?
Or does it only refer to "base" cFS  (cFS folder) and so the Simulators system in the Architecture includes also the component apps?
I would like to know the actual communication flow from ground station to the components cFS app because in the NOS Engine Server terminal, by listing all the nodes, I see that on "usart_1" bus there's only "gps" and not "fsw".
Instead, on "usart_2" bus we can find "reaction wheel 0" and "fsw", as shown by the attached screenshot.

![nos_eng_server_1](https://github.com/nasa/nos3/assets/153757151/90c5794b-9b57-400a-90d6-ed6e75dab5b8)

Thank you so much for the support.

**Comment by jlucas9** [comment]
Answer captured in https://github.com/nasa/nos3/issues/235
